### DISCUSSION TITLE
ApplicationModuleTest resides in different package then module resulting in "Package %s is not part of any module!"

### DISCUSSION THREAD

**Discussion by marinusgeuze** [spring-projects/spring-modulith#413] [Q&A] [answered]
Hi,

In our project, we have organized our tests into different root packages based on their types, such as unit tests, integration tests, and end-to-end tests. Consequently, our Spring Boot Integration Tests are placed in a package different from their corresponding modules, leading to the occurrence of the “Package %s is not part of any module!” error.

To provide more context, our test package structure looks something like this:

```
unit.com.test..*UnitTest
it.com.test..*IntegrationTest
end2end.com.test..*End2EndTest
```

We are seeking a solution to explicitly define the module to which a Spring Boot Integration Test belongs. This as part of the `ApplicationModuleTest` annotation. Or ideally, we would like to add the `it.com.test` package to the module. Is there a way to achieve this?

We acknowledge that a possible recommendation might be to group tests within the module boundaries, like` com.test..unit.*UnitTest`, `com.test..it.*IntegrationTest`, etc. However, this is not a feasible option for us due to the configuration constraints of our test frameworks.

**Comment by odrotbohm** [comment, marked as answer]
This sounds like a reasonable enhancement. I've created #466 to keep track of this.

**Reply by marinusgeuze** [reply]
I'm pleased to learn that this issue will be addressed and resolved. Thanks.
